redis-hbo-provider/src/test/java/com/facebook/presto/statistic/TestHistoricalStatisticsSerde.java

```diff
@@ -94,7 +94,7 @@ public void TestPlanStatisticsList()
     }
 
     @Test
-    public void TestHistoricalStatisticsDecodeValueException()
+    public void testHistoricalStatisticsDecodeValueException()
```

Comment:
assertThrows is fine, but you can also use the test annotation instead since you're only checking one thing here
```
@Test(expectedExceptions = RuntimeException.class)
```
(is RuntimeException the specific kind of exception that gets thrown or is it something that extends RuntimeException? Better to be more specific if there is a more specific exception to make sure the error handling is as expected)
You can also add an expectedExceptionsMessageRegexp to the annotation to match on the expected error message.

Reply:
Thanks, added a check on the exception message. Currently we wrap the exception with a runtime one since HistoricalStatisticsSerde class is an implementation of an oss interface which expects the instance to process exceptions.

Reply:
(The below definitely does not need to be addressed here, just a suggestion in general about error handling for this plugin) 

You can consider wrapping in a PrestoException  instead of a plain RuntimeException. Then you can also add an ErrorCodeSupplier for this plugin with relevant error codes and error categories (You can check the codebase for examples e.g. HiveErrorCode, CassandraErrorCode, etc.).  That way any exceptions thrown from here can be easily understood and categorized for error tracking purposes if a user encounters them.  Otherwise I think any non-PrestoExceptions for a query ultimately get wrapped in a PrestoException with GENERIC_INTERNAL_ERROR as the error code.

Reply:
Thanks a lot for the detailed explanation, I will surely consider this doing in a follow up patch.